src/Cli/dotnet/Installer/Windows/Security/AuthentiCode.cs

```diff
@@ -0,0 +1,196 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Runtime.Versioning;
+using System.Security.Cryptography;
+using System.Security.Cryptography.Pkcs;
+using System.Security.Cryptography.X509Certificates;
+
+namespace Microsoft.DotNet.Installer.Windows.Security
+{
+    /// <summary>
+    /// Contains various utilities methods around verifying AuthentiCode signatures on Windows.
+    /// </summary>
+#if NETCOREAPP
+    [SupportedOSPlatform("windows")]
+#endif
+    internal static class AuthentiCode
```

Comment:
I only have a cursory understanding of signing, so this looks fine to me but I don't have the knowledge to give it a in depth review.